docs/user-guides/configurations.md

```diff
@@ -32,14 +32,16 @@ Self-optimizing configurations are applicable to both Iceberg Format and Mixed s
 | self-optimizing.enabled                             | true             | Enables Self-optimizing                                |
 | self-optimizing.group                               | default          | Optimizer group for Self-optimizing                                   |
 | self-optimizing.quota                               | 0.1              | Quota for Self-optimizing, indicating the CPU resource the table can take up                       |
-| self-optimizing.execute.num-retries                         | 5                | Number of retries after failure of Self-optimizing                       |
+| self-optimizing.execute.num-retries                 | 5                | Number of retries after failure of Self-optimizing                       |
 | self-optimizing.target-size                         | 134217728(128MB) | Target size for Self-optimizing                           |
 | self-optimizing.max-file-count                      | 10000            | Maximum number of files processed by a Self-optimizing process              |               |
```

Comment:
It seems `self-optimizing.max-file-count ` is no longer used, shoulde we remove it from docs too?

Reply:
This parameter still works when performing a keyed table scan and is used to limit the number of cached change store files to avoid OOM of AMS.

I haven't found a better way to replace this parameter yet.

Reply:
I found that `com.netease.arctic.server.optimizing.OptimizingConfig#getMaxFileCount` has never been called, so I mistakenly thought that this parameter was useless.

We can discuss in other issues/PRs about:

- whether to standardize the use of parameters in AMS
- whether to unify the configuration parameters of different formats.